internal/pkg/vm/actor/builtin/storagemarket/storagemarket_test.go

```diff
@@ -68,32 +68,16 @@ func TestStorageMarketCreateStorageMinerDoesNotOverwriteActorBalance(t *testing.
 
 	st, vms := th.RequireCreateStorages(ctx, t)
 
-	// create account of future miner actor by sending FIL to the predicted address
-	minerAddr, err := address.NewIDAddress(100) // 100 is the first address of a constructed actor
+	// attempt create account of future miner actor by sending FIL to the predicted address
+	minerAddr, err := address.NewIDAddress(102) // 102 is the first address of a constructed actor outside genesis
 	require.NoError(t, err)
 
 	msg := types.NewUnsignedMessage(address.TestAddress2, minerAddr, 0, types.NewAttoFILFromFIL(100), types.SendMethodID, []byte{})
-	result, err := th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
-	require.NoError(t, err)
-	require.Equal(t, uint8(0), result.Receipt.ExitCode)
-
-	pdata := actor.MustConvertParams(types.OneKiBSectorSize, th.RequireRandomPeerID(t))
-	msg = types.NewUnsignedMessage(address.TestAddress, address.StorageMarketAddress, 0, types.NewAttoFILFromFIL(200), storagemarket.CreateStorageMiner, pdata)
-	result, err = th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
-	require.NoError(t, err)
-	require.Equal(t, uint8(0), result.Receipt.ExitCode)
-	require.NoError(t, result.ExecutionError)
-
-	// ensure our derived address is the address storage market creates
-	derivedAddr, err := deriveMinerAddress(address.TestAddress, 0)
-	createdAddress, err := address.NewFromBytes(result.Receipt.Return[0])
-	require.NoError(t, err)
-	assert.Equal(t, derivedAddr, createdAddress)
-	miner, err := st.GetActor(ctx, minerAddr)
-	require.NoError(t, err)
+	_, err = th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
 
-	// miner balance should be sum of messages
-	assert.Equal(t, types.NewAttoFILFromFIL(300).String(), miner.Balance.String())
+	// message fails because we do not create actors when message to is an ID address.
```

Comment:
Is this true with the given code?  I can't find the account actor init state function (which I think is supposed to raise the error) getting called from the `getOrCreateActor` call.

Reply:
It's thrown from https://github.com/filecoin-project/go-filecoin/pull/3700/files#diff-b56a91c5cc23ee17bbf77b69af05eae5R689 when we resolve the address as an ID address and then attempt to get it from the state tree directly.